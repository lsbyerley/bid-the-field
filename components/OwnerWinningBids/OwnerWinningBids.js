import { getOwnerWinningBids, getPlayerFromBid } from '../../util/auctionUtils';

const OwnerWinningBids = ({ bids = [], session = {}, playersData }) => {
  const winningBids = getOwnerWinningBids(bids, session?.user?.email);
  const totalBidAmount = winningBids.reduce(
    (previousValue, currentValue) => previousValue + currentValue.amount,
    0
  );

  return (
    <div className='card bg-base-200'>
      <div className='card-body'>
        <div className='card-title'>Your Winning Bids ${totalBidAmount}</div>
        <ul role='list' className='divide-y divide-gray-200'>
          {winningBids.map((bid) => (
            <li key={bid.id} className='px-6 py-4'>
              <span>${bid.amount}</span>
              <span className='ml-4'>
                {getPlayerFromBid(playersData, bid.player_id)}
              </span>
            </li>
          ))}
          {!winningBids ||
            (!winningBids?.length && (
              <li className='px-6 py-4'>
                <span className='text-red-500'>no winning bids</span>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
};

export default OwnerWinningBids;
