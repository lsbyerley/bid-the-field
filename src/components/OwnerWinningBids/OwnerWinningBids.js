import { getOwnerWinningBids, getPlayerFromBid } from '../../lib/auctionUtils';
import useAsyncReference from '../../lib/useAsyncReference';

const OwnerWinningBids = ({ bids = [], session = {}, playersData }) => {
  const asyncBids = useAsyncReference(bids, true);
  const winningBids = getOwnerWinningBids(
    asyncBids.current,
    session?.user?.email
  );
  const totalBidAmount = winningBids.reduce(
    (previousValue, currentValue) => previousValue + currentValue.amount,
    0
  );

  return (
    <div className='rounded-lg card card-compact bg-base-200'>
      <div className='card-body'>
        <div>
          <h3 className='text-lg font-medium'>
            <span className='mr-2 text-sm'>Your Winning Bids:</span>
            <span className='text-success'>
              ${Number.parseFloat(totalBidAmount).toFixed(2)}
            </span>
          </h3>
          <dl className='mt-2 border-t border-b border-gray-200 divide-y divide-gray-200'>
            {winningBids.map((bid) => (
              <div
                key={bid.id}
                className='flex justify-between py-3 text-sm font-medium'
              >
                <dt className=''>
                  {getPlayerFromBid(playersData, bid.player_id)}
                </dt>
                <dd className=''>${bid.amount}</dd>
              </div>
            ))}
            {!winningBids ||
              (!winningBids?.length && (
                <div className='px-2 py-3'>
                  <dt className='text-warning'>no winning bids</dt>
                </div>
              ))}
          </dl>
        </div>
      </div>
    </div>
  );
};

export default OwnerWinningBids;
