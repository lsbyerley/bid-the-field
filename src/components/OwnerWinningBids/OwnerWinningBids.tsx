import { getOwnerWinningBids, getPlayerFromBid } from '@/lib/auctionUtils';
import useAsyncReference from '@/lib/useAsyncReference';

import type { OwnerWinningBidsProps, Bid } from '@/types';

const OwnerWinningBids = ({
  bids = [],
  user,
  players = [],
}: OwnerWinningBidsProps) => {
  const asyncBids = useAsyncReference(bids, true);
  const winningBids = getOwnerWinningBids(asyncBids.current, user?.id);
  const totalBidAmount: number = winningBids.reduce(
    (previousValue: number, currentValue: Bid) =>
      previousValue + currentValue.amount,
    0
  );

  return (
    <div className='rounded-lg md:row-span-3 card card-compact bg-base-100'>
      <div className='card-body'>
        <div>
          <div className='flex justify-between pb-2 mb-4 text-lg font-medium border-b'>
            <span className='mr-2 truncate text-info'>Your Winning Bids</span>
            <span className='text-success'>${totalBidAmount.toFixed(2)}</span>
          </div>

          <dl className='mt-2 overflow-y-scroll divide-y max-h-72'>
            {winningBids.map((bid: Bid) => {
              const player = getPlayerFromBid(players, bid.player_id);
              return (
                <div
                  key={bid.id}
                  className='flex justify-between py-3 text-sm font-medium'
                >
                  <dt className=''>
                    {player.seed && `(${player.seed}) `}
                    {player.full_name ||
                      player.short_name ||
                      `na:${bid.player_id}`}
                  </dt>
                  <dd className=''>${bid.amount.toFixed(2)}</dd>
                </div>
              );
            })}
            {!winningBids ||
              (!winningBids?.length && (
                <div className='px-2 py-3'>
                  <dt className='text-warning'>no winning bids</dt>
                </div>
              ))}
          </dl>
        </div>
      </div>
    </div>
  );
};

export default OwnerWinningBids;
