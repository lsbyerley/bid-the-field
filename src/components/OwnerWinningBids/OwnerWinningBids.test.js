import React from 'react';
import { render, screen } from 'test-utils';
import OwnerWinningBids from './OwnerWinningBids';

jest.mock('@/lib/auctionUtils', () => ({
  getPlayerFromBid: jest.fn(() => {
    return {};
  }),
  getOwnerWinningBids: jest.fn(() => {
    return [];
  }),
}));

const renderOwnerWinningBids = (props) => {
  const utils = render(<OwnerWinningBids {...props} />);
  const home = utils.container.firstChild;
  return { ...utils, home };
};

describe('OwnerWinningBids', () => {
  it('should render and match snapshot', () => {
    const { home } = renderOwnerWinningBids();
    expect(home).toBeTruthy();
    expect(home).toMatchSnapshot();
  });
});
