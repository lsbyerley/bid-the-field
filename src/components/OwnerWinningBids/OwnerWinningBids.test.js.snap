// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OwnerWinningBids should render and match snapshot 1`] = `
<div
  class="rounded-lg md:row-span-3 card card-compact bg-base-100"
>
  <div
    class="card-body"
  >
    <div>
      <div
        class="flex justify-between pb-2 mb-4 text-lg font-medium border-b"
      >
        <span
          class="mr-2 truncate text-info"
        >
          Your Winning Bids
        </span>
        <span
          class="text-success"
        >
          $
          0.00
        </span>
      </div>
      <dl
        class="mt-2 overflow-y-scroll divide-y max-h-72"
      >
        <div
          class="px-2 py-3"
        >
          <dt
            class="text-warning"
          >
            no winning bids
          </dt>
        </div>
      </dl>
    </div>
  </div>
</div>
`;
